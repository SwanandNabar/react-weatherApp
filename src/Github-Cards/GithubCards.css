.cards {
    justify-content: space-between;
    max-width: 90%;
    margin: auto;
    flex: 1;
}

.card {
    list-style-type: none;
    display: inline;
    padding-left: 15px;
    padding-right: 15px;
}

.card_inner {
    padding: 10px;
    height: 100%;
    box-shadow:  0 0 15px -2px rgba(0,0,0,.3);
}

.centered {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
}

.action_buttons {
    position: relative;
    align-items: center;
    display: inline;
    padding: 10px;
}

.repos_list {
    border: 3px solid transparent;
    border-radius: 5px;
    height: 50px;
    cursor: pointer;
    align-items: center;
    padding: 10px 20px;
    line-height: 1.1;
    vertical-align: middle;
    background-color: #4a58b0;
    color: #fff;
}

.card_header {
    background-color: #29b9bb;
    padding: 10px;
    color: #fff;
    text-align: center;
    cursor: pointer;
}

.card_info {
    border-top : 1px solid #bebec1;
    margin: 10px;
}

.commit_list {
    border-top : 1px solid #bebec1;
    margin: 10px;
    padding-top: 5px;
}