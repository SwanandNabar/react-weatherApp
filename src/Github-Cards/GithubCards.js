import React, { Component } from 'react';
import axios from 'axios';
import * as moment from 'moment';
import './GithubCards.css';
import 'babel-polyfill';

export default class GithubCards extends Component {

    constructor(props) {
        super(props);
        this.state = {
            topHundredRepos: [],
            checkBox: false,
            commitDetails: []
        }
        this.getRepos = this.getRepos.bind(this);
        this.getCommitList = this.getCommitList.bind(this);
        this.getCommitListRepo = this.getCommitListRepo.bind(this);
    }

    async getRepos() {
        // construct URL for top 100 starred repos
        var BASE_URL = `https://api.github.com`;
        var SEARCH_REPO = `/search/repositories?q=stars:>10000&sort=stars&order=desc&per_page=100`;
        var TOP_HUNDRED_URL = BASE_URL + SEARCH_REPO;

        axios.get(TOP_HUNDRED_URL)
            .then(result => {
                result.data.items.map(item => {
                    this.setState({
                        topHundredRepos: this.state.topHundredRepos.concat({
                            "name": item.name,
                            "url": item.url,
                            "starCount": item.stargazers_count,
                            "loginName": item.owner.login
                        })
                    })
                })
            }).catch(e => {
                console.error(e)
            })
    }

    getCommitList() {
        if (!this.state.checkBox) {
            this.setState({
                checkBox: true
            })
            // construct URL for list of commits on those repos
            let timeOneDayBefore = moment.utc().subtract(24, "hours").format("YYYY-MM-DDTHH:mm:ss");

            var BASE_URL = `https://api.github.com`;
            this.state.topHundredRepos.map(item => {
                axios.get(BASE_URL + `/repos/${item.loginName}/${item.name}/commits?since=${timeOneDayBefore}`)
                    .then(result => { })
                    .catch(e => {
                        console.error(e);
                    })
            })
        } else if (this.state.checkBox) {
            this.setState({
                checkBox: false
            })
        }
    }

    getCommitListRepo() {
        let timeOneDayBefore = moment.utc().subtract(24, "hours").format("YYYY-MM-DDTHH:mm:ss");
        var BASE_URL = `https://api.github.com`;
        this.state.topHundredRepos.map(item => {
            axios.get(BASE_URL + `/repos/${item.loginName}/${item.name}/commits?since=${timeOneDayBefore}`)
                .then(result => { })
                .catch(e => {
                    console.error(e);
                })
        })

    }

    render() {
        return (
            <div className="centered">
                <div className="action_buttons">
                    <button className="repos_list" onClick={this.getRepos}>Get repos</button>
                    <span>Get the list of commits in last 24 hours</span>
                    <input className="check_box"
                        type="checkbox"
                        onChange={this.getCommitList} />
                </div>
                {this.state.topHundredRepos.map((repo, index) =>
                    <ul className="cards" key={index}>
                        <li key={repo.name} className="card">
                            <section className="card_inner">
                                <div className="card_header">
                                    <h3 id={repo.name} onClick={this.getCommitListRepo}>{repo.name}</h3>
                                </div>
                                <div className="card_info">
                                    <p>GitHub URL : {repo.url}</p>
                                    <p>Star Count: {repo.starCount}</p>
                                </div>
                                {this.state.checkBox &&
                                    <div className="commit_list">
                                        here is the list of commits
                                    </div>}
                            </section>
                        </li>
                    </ul>
                )}
            </div>
        )
    }
}
