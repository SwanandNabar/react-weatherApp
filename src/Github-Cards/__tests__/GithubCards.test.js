import GithubCards from '../GithubCards';

let githubCards;
describe ("GithubCards", () => {
    beforeEach(() => {
        githubCards = new GithubCards();
    });

    it ("Loads the component", () => {
        expect(githubCards).toMatchSnapshot();
    });

    it("Calls the getRepos function", () => {
        githubCards.getRepos();
        expect(githubCards).toMatchSnapshot();
    });

    it("Calls the getCommitList function", () => {
        githubCards.getCommitList();
        expect(githubCards).toMatchSnapshot();
    });

    it("Calls the getCommitListRepo function", () => {
        githubCards.getCommitListRepo();
        expect(githubCards).toMatchSnapshot();
    })
});
