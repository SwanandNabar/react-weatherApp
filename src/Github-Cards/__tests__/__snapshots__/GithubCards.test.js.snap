// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubCards Calls the getCommitList function 1`] = `
GithubCards {
  "context": undefined,
  "getCommitList": [Function],
  "getCommitListRepo": [Function],
  "getRepos": [Function],
  "props": undefined,
  "refs": Object {},
  "state": Object {
    "checkBox": false,
    "commitDetails": Array [],
    "topHundredRepos": Array [],
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;

exports[`GithubCards Calls the getCommitListRepo function 1`] = `
GithubCards {
  "context": undefined,
  "getCommitList": [Function],
  "getCommitListRepo": [Function],
  "getRepos": [Function],
  "props": undefined,
  "refs": Object {},
  "state": Object {
    "checkBox": false,
    "commitDetails": Array [],
    "topHundredRepos": Array [],
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;

exports[`GithubCards Calls the getRepos function 1`] = `
GithubCards {
  "context": undefined,
  "getCommitList": [Function],
  "getCommitListRepo": [Function],
  "getRepos": [Function],
  "props": undefined,
  "refs": Object {},
  "state": Object {
    "checkBox": false,
    "commitDetails": Array [],
    "topHundredRepos": Array [],
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;

exports[`GithubCards Loads the component 1`] = `
GithubCards {
  "context": undefined,
  "getCommitList": [Function],
  "getCommitListRepo": [Function],
  "getRepos": [Function],
  "props": undefined,
  "refs": Object {},
  "state": Object {
    "checkBox": false,
    "commitDetails": Array [],
    "topHundredRepos": Array [],
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
